docs/architecture/grpc-for-wcf-developers/migrate-duplex-services.md

```diff
@@ -158,17 +158,17 @@ public class StockTickerService : Protos.SimpleStockTicker.SimpleStockTickerBase
 }
 ```
 
-As you can see, although the declaration in the `.proto` file says the method "returns" a stream of `StockTickerUpdate` messages, in fact it returns a vanilla `Task`. The job of creating the stream is handled by the generated code and the gRPC runtime libraries, which provide the `IServerStreamWriter<StockTickerUpdate>` response stream, ready to use.
+As you can see, although the declaration in the `.proto` file says the method returns a stream of `StockTickerUpdate` messages, it actually returns a generic `Task`. The job of creating the stream is handled by the generated code and the gRPC runtime libraries, which provide the `IServerStreamWriter<StockTickerUpdate>` response stream, ready to use.
 
-Unlike a WCF duplex service, where the instance of the service class is kept alive while the connection is open, the gRPC service uses the returned Task to keep the service alive. The Task shouldn't complete until the connection is closed.
+Unlike a WCF duplex service, where the instance of the service class is kept alive while the connection is open, the gRPC service uses the returned task to keep the service alive. The task shouldn't complete until the connection is closed.
 
-The service can tell when the client has closed the connection by using the `CancellationToken` from the `ServerCallContext`. A simple static method, `AwaitCancellation`, is used to create a Task that completes when the token is canceled.
+The service can tell when the client has closed the connection by using the `CancellationToken` from the `ServerCallContext`. A simple static method, `AwaitCancellation`, is used to create a task that completes when the token is canceled.
 
-In the `Subscribe` method, then, get a `StockPriceSubscriber` and add an event handler that writes to the response stream. Then wait for the connection to be closed, before immediately disposing the `subscriber` to prevent it trying to write data to the closed stream.
+In the `Subscribe` method, then, get a `StockPriceSubscriber` and add an event handler that writes to the response stream. Then wait for the connection to be closed before immediately disposing the `subscriber` to prevent it from trying to write data to the closed stream.
 
-The `WriteUpdateAsync` method has a `try`/`catch` block to handle any errors that might happen when writing a message to the stream. This is an important consideration in persistent connections over networks, which could be broken at any millisecond, whether intentionally or because of a failure somewhere.
+The `WriteUpdateAsync` method has a `try`/`catch` block to handle any errors that might happen when it writes a message to the stream. This consideration is important in persistent connections over networks, which could be broken at any millisecond, whether intentionally or because of a failure somewhere.
```

Comment:
Sounds weird. `when "it" writes a message` definitely has an ambiguous pronoun here. `WriteUpdateAsync` is not the thing doing the writing, it is merely called by whoever it writing.

Reply:
fixed